frontend/src/components/maps/leaflet/MapProperties.test.tsx

```diff
@@ -158,96 +162,73 @@ describe('MapProperties View', () => {
         </Router>
       </Provider>,
     );
-    await act(() => (promise as unknown) as Promise<void>);
+    wait(() => expect(mapRef.current).toBeDefined(), { timeout: 500 });
```

Comment:
what's the difference between wait() and await act()?

Reply:
wait is used to allow an expectation to finish before continuing. Behind the scenes the wait runs in a loop, and just ignores any failures or errors until the timeout is reached. 

act is used to wrap an action that will cause a change in a rendered component. For example, clicking a submit button. Sometimes the act will also need to check for something to be true before exiting.

Generally use wait if you are waiting for something that isn't a direct result of one of your actions. Use act otherwise.